### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by mpa-mxiang**
For this PR, I:

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it to the upcoming projects.
- Create forms to perform the following functions:
- Create a Post on behalf of the current_user (use the method you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
```
Comment (codecaiine) [inline app/controllers/comments_controller.rb:2]: - [x] According to the project requieremt , We should have **new and create actions** in comment controller as you did in Post controller 👇 : Please make sure to add the missing **new action** to make the user able to call the comment form require for this project and make your project as expected 👍 

- [x] According to the project requirement, you need to create a form in the **views/comments/new.html.erb** file with inputs to make user able to create comments. 😄

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      post 'like', on: :member
+      resources :comments, only: [:create]
```
Comment (codecaiine) [inline config/routes.rb]: - [x] According to project requirement you should have a pair of route for creating comments as you can see below 👇  . Please make sure to add the missing routes in order to make your project works as expected 👍 

 ```
   POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

File: config/routes.rb [hunk 3]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      post 'like', on: :member
```
Comment (codecaiine) [inline config/routes.rb:5]: - [x] Good job so far But you missed to create a controller file with create actions to make user able to add a like to a post . Please fix it 👍

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,10 +1,46 @@
...
+      existing_like.destroy
+      @post.decrement!(:likes_counter)
+    else
+      current_user.likes.create(post: @post)
+    end
+
+    redirect_to user_post_path(@post.author, @post)
+  end
+
+  def new
```
Comment (codecaiine) [inline app/controllers/posts_controller.rb:23]: Good job so far but I think you should have create in the **views/posts/new.html.erb** file which accepts input to make user to be able to create posts. Please fix them 👍

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @mpa-mxiang , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good form setup ✔️ 
- post routes are well setup ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (mpa-mxiang) [reply app/controllers/posts_controller.rb:23]: I have this one called `_new.html.erb` located in the `./app/views/users`. @codecaiine  I am not sure what should I add here, could you please clarify more?

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @mpa-mxiang 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- Now User can add a Cooment and a Like :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mpa-mxiang** [comment]
Thank you @tresorsawasawa for the approval

**Pull request merged**